doc/YamlConfigFile.md

```diff
@@ -0,0 +1,298 @@
+# YAML configuration file
+
+This page describes the configuration file, which controls the behavior of the Windows service.
+
+You can find configuration file samples in the [examples](../examples) directory of the source code repository.
+Actual samples are being also published as a part of releases on GitHub and NuGet.
+
+## File structure
+
+YAML Configuration file shuold be in following format
+
+Example:
+
+```yaml
+id: jenkins
+name: Jenkins
+description: This service runs Jenkins continuous integration system.
+env:
+    - name: JENKINS_HOME
+      value: '%BASE%'
+executable: java
+arguments: >
+    -Xrs
+    -Xmx256m
+    -jar "%BASE%\jenkins.war"
+    --httpPort=8080
+log:
+    mode: roll
+```
+
+## Environment variable expansion
+
+Configuration YAML files can include environment variable expansions of the form `%Name%`. 
+Such occurrences, if found, will be automatically replaced by the actual values of the variables. [Read more. . . ](xmlConfigFile.md#environment-variable-expansion)
+
+## Configuration entries
+
+### id
```

Comment:
There is a lot of duplication between this doc and XML. Maybe we should move detailed docs into a separate doc and link them.

It does not block this PR, can be done later. Docs need update for v3 anyway

Reply:
Indeed. I am also thinking its more user friendly. Anyway if we move into `winsw.github.io` then we can present that in much more user friendly way.